infra/src/main/java/com/movie/infra/ratelimit/GuavaRateLimitService.java

```diff
@@ -0,0 +1,74 @@
+package com.movie.infra.ratelimit;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.util.concurrent.RateLimiter;
+import com.movie.common.exception.RateLimitExceededException;
+import com.movie.common.service.RateLimitService;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import java.util.concurrent.TimeUnit;
+
+@Service
+@Profile("!prod")
+public class GuavaRateLimitService implements RateLimitService {
+    private final Cache<String, RateLimiter> rateLimiters;
+    private final Cache<String, Integer> requestCounts;
+    private final Cache<String, RateLimiter> reservationRateLimiters;
+    private final int maxRequestsPerMinute = 50;
+
+    public GuavaRateLimitService() {
+        this.rateLimiters = CacheBuilder.newBuilder()
+                .expireAfterWrite(1, TimeUnit.HOURS)
+                .build();
+
+        this.requestCounts = CacheBuilder.newBuilder()
+                .expireAfterWrite(1, TimeUnit.MINUTES)
+                .build();
+
+        this.reservationRateLimiters = CacheBuilder.newBuilder()
+                .expireAfterWrite(5, TimeUnit.MINUTES)
+                .build();
+    }
+
+    @Override
+    public void checkIpRateLimit(String ip) {
+        Integer count = requestCounts.getIfPresent(ip);
+        if (count != null && count >= maxRequestsPerMinute) {
+            throw new RateLimitExceededException("IP가 차단되었습니다. 1시간 후에 다시 시도해주세요.");
+        }
+
+        RateLimiter limiter = rateLimiters.getIfPresent(ip);
+        if (limiter == null) {
+            limiter = RateLimiter.create(maxRequestsPerMinute / 60.0); // 초당 요청 수로 변환
+            rateLimiters.put(ip, limiter);
+        }
```

Comment:
그냥 get을 사용하는건 어때요? 동시성 때문에 너도나도 새로운 RateLimiter를 만들 수도 있어요.

```suggestion
        RateLimiter limiter = rateLimiters.get(ip, () -> RateLimiter.create(maxRequestsPerMinute / 60.0));
```